pkg/dependencies/dependents.go

```diff
@@ -0,0 +1,194 @@
+		ctx:       ctx,
+	}
+}
+
+func (deps *dependencies) GetSortedDependents() ([]PackageName, error) {
+	packages, err := findAllDependents(deps.gqlClient)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get dependents: %v", err)
+	}
+
+	var packagesArr []PackageName
+
+	for n, d := range packages {
+		packagesArr = append(packagesArr, PackageName{Name: n, DependentCount: len(d.dependents)})
+	}
+
+	sort.Slice(packagesArr, func(i, j int) bool {
+		return packagesArr[i].DependentCount > packagesArr[j].DependentCount
+	})
+	return packagesArr, nil
+}
+
+// findAllDependents queries a GraphQL endpoint to find all dependencies and constructs a map of dependencyNode.
+func findAllDependents(gqlClient graphql.Client) (map[string]dependencyNode, error) {
+	ctx := context.Background()
+
+	// Initialize a map to hold package names mapped to their dependency nodes. This will be the returned map
+	packages := map[string]dependencyNode{}
+
+	// Initialize maps to hold dependency and dependent edges.
+	// dependentEdges maps a package ID to the IDs of packages that depend on it (DepPkg to Pkg).
+	// dependencyEdges maps a package ID to the IDs of its dependencies (Pkg to DepPkg).
+	dependentEdges := make(map[string][]string)  // id -> dependents ids
+	dependencyEdges := make(map[string][]string) // id -> dependency ids
+
+	// Initialize a map to convert package IDs to their names.
+	idToName := make(map[string]string)
+
+	sboms, err := model.HasSBOMs(ctx, gqlClient, model.HasSBOMSpec{})
+
+	if err != nil {
+		return nil, fmt.Errorf("error getting dependencies: %v", err)
+	}
+
+	for _, resp := range sboms.HasSBOM {
+		// Skip entries from "deps.dev" because they are inconsistent.
+		if resp.Origin == deps_dev.DepsCollector {
+			continue
+		}
+		// Iterate through the included dependencies of each SBOM.
+		for _, dependency := range resp.IncludedDependencies {
+			// Construct unique names for the dependency package and the package itself.
+			// TODO: Make the names actually unique, not just add "_".
+			depPkgName := dependency.DependencyPackage.Type + "_" + dependency.DependencyPackage.Namespaces[0].Namespace + "_" + dependency.DependencyPackage.Namespaces[0].Names[0].Name
+			pkgName := dependency.Package.Type + "_" + dependency.Package.Namespaces[0].Namespace + "_" + dependency.Package.Namespaces[0].Names[0].Name
+
+			var depPkgIds []string
+			pkgId := dependency.Package.Namespaces[0].Names[0].Versions[0].Id
+
+			if len(dependency.DependencyPackage.Namespaces[0].Names[0].Versions) == 0 {
+				findMatchingDepPkgVersionIDs, err := cmd.FindDepPkgVersionIDs(ctx, gqlClient, dependency.DependencyPackage.Type,
+					dependency.DependencyPackage.Namespaces[0].Namespace,
+					dependency.DependencyPackage.Namespaces[0].Names[0].Name, dependency.VersionRange)
+				if err != nil {
+					return nil, fmt.Errorf("error from FindMatchingDepPkgVersionIDs:%w", err)
+				}
+				depPkgIds = append(depPkgIds, findMatchingDepPkgVersionIDs...)
```

Comment:
no there if the version is an empty string, it would use the version range:
https://github.com/guacsec/guac/blob/431a2863ffd7ceef08e48661611704803dba0ab0/pkg/ingestor/parser/common/helpers.go#L119-L124